### PR TITLE
Added remote upgrading for MacOS agents

### PR DESCRIPTION
**Pull request by cborla**
|Related issue|
|---|
|https://github.com/wazuh/wazuh-documentation/issues/4130|

## Description
Support for remote upgrading of MacOS agents needs to be documented.

## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar. 
- [x] Used impersonal speech. 
- [x] Used uppercase only on nouns. 
- [x] Updated the `redirect.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).

## Note to the reviewer

This PR includes changes to the `redirect.js` script that need to be included in all production branches.

### CODE REVIEW

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 1]
```diff
@@ -131,6 +131,69 @@ Compile the WPK package using the MSI package and, your SSL certificate and key:
 
   # tools/agent-upgrade/wpkpack.py output/myagent.wpk path/to/wpkcert.pem path/to/wpkcert.key path/to/wazuhagent.msi path/to/upgrade.bat path/to/do_upgrade.ps1
 
+MacOS WPK
+^^^^^^^^^
+
+Install development tools and compilers. In macOS, this can be easily done by installing brew, a package manager for macOS:
```
Comment (TomasTurina) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:137]: This is not OK. MacOS uses `.pkg` file, no compiled files.

File: source/user-manual/agents/remote-upgrading/wpk-list.rst [hunk 2]
```diff
@@ -22,3 +22,12 @@ Windows
...
+| Distribution | Version          | Architecture | WPK Package                                                                                                                                                                                                                                                          |
++==============+==================+==============+======================================================================================================================================================================================================================================================================+
+|    MacOS     | |WAZUH_LATEST|            |    64bit     | `wazuh_agent_v|WAZUH_LATEST|_macos_x86_64.wpk <https://packages.wazuh.com/4.x/wpk/macos/x86_64/wazuh_agent_v|WAZUH_LATEST|_macos_x86_64.wpk>`_ (`sha512 <https://packages.wazuh.com/|CURRENT_MAJOR|/checksums/wazuh/|WAZUH_LATEST|/wazuh_agent_v|WAZUH_LATEST|_linux_x86_64.wpk.sha512>`__)                          |
```
Comment (TomasTurina) [inline source/user-manual/agents/remote-upgrading/wpk-list.rst:32]:
```suggestion
|    MacOS     | |WAZUH_LATEST|            |    64bit     | `wazuh_agent_v|WAZUH_LATEST|_macos_x86_64.wpk <https://packages.wazuh.com/4.x/wpk/macos/x86_64/pkg/wazuh_agent_v|WAZUH_LATEST|_macos_x86_64.wpk>`_ (`sha512 <https://packages.wazuh.com/|CURRENT_MAJOR|/checksums/wazuh/|WAZUH_LATEST|/wazuh_agent_v|WAZUH_LATEST|_macos_x86_64.wpk.sha512>`__)                          |
```

**Review by TomasTurina — CHANGES_REQUESTED** [review]

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst (hunk 1 above)
Comment (cborla) [reply source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst]: Done

File: source/user-manual/agents/remote-upgrading/wpk-list.rst (hunk 2 above)
Comment (cborla) [reply source/user-manual/agents/remote-upgrading/wpk-list.rst]: Done

**Review by TomasTurina — APPROVED** [review]

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 3]
```diff
@@ -89,6 +90,13 @@ Compile the WPK package using your SSL certificate and key:
 
 In this example, the Wazuh project's root directory contains the proper ``upgrade.sh`` file.
 
+Definitions:
+    - ``output/myagent.wpk`` is the name of the output WPK package.
+    - ``path/to/wpkcert.pem`` is the path to SSL certificate.
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:95]:
```suggestion
    - ``path/to/wpkcert.pem`` is the path to the SSL certificate.
```

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 4]
```diff
@@ -89,6 +90,13 @@ Compile the WPK package using your SSL certificate and key:
...
+    - ``path/to/wpkcert.key`` is the path to SSL certificate's key.
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:96]:
```suggestion
    - ``path/to/wpkcert.key`` is the path to the SSL certificate's key.
```

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 5]
```diff
@@ -89,6 +90,13 @@ Compile the WPK package using your SSL certificate and key:
...
+    - ``\*`` is the file (or the files) to be included into WPK package. In this case, all the contents will be added.
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:97]:
```suggestion
    - ``\*`` is the file or files to be included in the WPK package. In this case, all the contents are added.
```

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 6]
```diff
@@ -133,12 +141,63 @@ Compile the WPK package using the MSI package and, your SSL certificate and key:
 
 Definitions:
     - ``output/myagent.wpk`` is the name of the output WPK package.
-    - ``path/to/wpkcert.pem`` is the path to your SSL certificate.
-    - ``path/to/wpkcert.key`` is the path to your SSL certificate's key.
-    - ``path/to/upgrade.bat`` is the path to the upgrade.bat file you can find an example at src/win32 within the wazuh repository or write your own.
-    - ``path/to/do_upgrade.ps1`` is the path to the do_upgrade.ps1 file you can find an example at src/win32 within the wazuh repository or write your own.
-    - ``path/to/wazuhagent.msi`` is the path to the MSI you have downloaded in step 3.
-    - ``\*`` is the file (or the files) to be included into the WPK package. In this case, all the contents will be added.
+    - ``path/to/wpkcert.pem`` is the path to SSL certificate.
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:144]:
```suggestion
    - ``path/to/wpkcert.pem`` is the path to the SSL certificate.
```

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 7]
```diff
@@ -133,12 +141,63 @@ Compile the WPK package using the MSI package and, your SSL certificate and key:
...
+    - ``path/to/wpkcert.key`` is the path to SSL certificate's key.
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:145]:
```suggestion
    - ``path/to/wpkcert.key`` is the path to the SSL certificate's key.
```

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 8]
```diff
@@ -133,12 +141,63 @@ Compile the WPK package using the MSI package and, your SSL certificate and key:
...
+    - ``path/to/wazuhagent.msi`` is the path to MSI file, downloaded in step 3.
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:146]:
```suggestion
    - ``path/to/wazuhagent.msi`` is the path to the MSI file downloaded in step 3.
```

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 9]
```diff
@@ -133,12 +141,63 @@ Compile the WPK package using the MSI package and, your SSL certificate and key:
...
+    - ``path/to/upgrade.bat`` is the path to the upgrade.bat file. Find an example at src/win32 within wazuh repository.
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:147]:
```suggestion
    - ``path/to/upgrade.bat`` is the path to the upgrade.bat file. Find an example in src/win32 in the Wazuh repository.
```

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 10]
```diff
@@ -133,12 +141,63 @@ Compile the WPK package using the MSI package and, your SSL certificate and key:
...
+    - ``path/to/do_upgrade.ps1`` is the path to the do_upgrade.ps1 file. Find an example at src/win32 within wazuh repository.
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:148]:
```suggestion
    - ``path/to/do_upgrade.ps1`` is the path to the do_upgrade.ps1 file. Find an example in src/win32 in the Wazuh repository.
```

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 11]
```diff
@@ -133,12 +141,63 @@ Compile the WPK package using the MSI package and, your SSL certificate and key:
...
+
+  # yum install make gcc policycoreutils-python automake autoconf libtool unzip
+
+For Debian-based distributions:
+
+.. code-block:: console
+
+  # apt-get install make gcc libc6-dev curl policycoreutils automake autoconf libtool unzip
+
+Download and extract the latest version of wazuh sources:
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:168]:
```suggestion
Download and extract the latest version of Wazuh sources:
```

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 12]
```diff
@@ -133,12 +141,63 @@ Compile the WPK package using the MSI package and, your SSL certificate and key:
...
+
+.. code-block:: console
+
+  # curl -Ls https://github.com/wazuh/wazuh/archive/v|WAZUH_LATEST|.tar.gz | tar zx
+
+Download the latest version of the wazuh PKG package:
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:174]:
```suggestion
Download the latest version of the Wazuh PKG package:
```

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 13]
```diff
@@ -133,12 +141,63 @@ Compile the WPK package using the MSI package and, your SSL certificate and key:
...
+Compile the WPK package using the PKG package and, your SSL certificate and key:
+
+.. code-block:: console
+
+  # tools/agent-upgrade/wpkpack.py output/myagent.wpk path/to/wpkcert.pem path/to/wpkcert.key path/to/wazuhagent.pkg path/to/upgrade.sh path/to/pkg_installer_mac.sh
+
+
+Definitions:
+    - ``output/myagent.wpk`` is the name of the output WPK package.
+    - ``path/to/wpkcert.pem`` is the path to SSL certificate.
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:196]:
```suggestion
    - ``path/to/wpkcert.pem`` is the path to the SSL certificate.
```

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 14]
```diff
@@ -133,12 +141,63 @@ Compile the WPK package using the MSI package and, your SSL certificate and key:
...
+    - ``path/to/wpkcert.key`` is the path to SSL certificate's key.
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:197]:
```suggestion
    - ``path/to/wpkcert.key`` is the path to the SSL certificate's key.
```

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 15]
```diff
@@ -133,12 +141,63 @@ Compile the WPK package using the MSI package and, your SSL certificate and key:
...
+    - ``path/to/wazuhagent.pkg`` is the path to PKG file, downloaded in step 3.
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:198]:
```suggestion
    - ``path/to/wazuhagent.pkg`` is the path to the PKG file downloaded in step 3.
```

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 16]
```diff
@@ -133,12 +141,63 @@ Compile the WPK package using the MSI package and, your SSL certificate and key:
...
+    - ``path/to/upgrade.sh`` is the path to the upgrade.sh file. Find an example at base directory within wazuh repository.
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:199]:
```suggestion
    - ``path/to/upgrade.sh`` is the path to the upgrade.sh file. Find an example at the base directory in the Wazuh repository.
```

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 17]
```diff
@@ -133,12 +141,63 @@ Compile the WPK package using the MSI package and, your SSL certificate and key:
...
+    - ``path/to/pkg_installer_mac.sh`` is the path to the pkg_installer_mac.sh file. Find an example at src/init within wazuh repository.
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:200]:
```suggestion
    - ``path/to/pkg_installer_mac.sh`` is the path to the pkg_installer_mac.sh file. Find an example in src/init in the Wazuh repository.
```

File: source/user-manual/agents/remote-upgrading/wpk-list.rst [hunk 18]
```diff
@@ -22,3 +22,12 @@ Windows
 +==============+=========+==============+===========================================================================================================================================================================================================================================================+
 |   Windows    |  |WAZUH_LATEST|  |   32/64bit   | `wazuh_agent_v|WAZUH_LATEST|_windows.wpk <https://packages.wazuh.com/4.x/wpk/windows/wazuh_agent_v|WAZUH_LATEST|_windows.wpk>`_ (`sha512 <https://packages.wazuh.com/|CURRENT_MAJOR|/checksums/wazuh/|WAZUH_LATEST|/wazuh_agent_v|WAZUH_LATEST|_windows.wpk.sha512>`__)                                   |
 +--------------+---------+--------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+
+MacOS
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/wpk-list.rst:26]:
```suggestion
macOS
```

File: source/user-manual/agents/remote-upgrading/wpk-list.rst [hunk 19]
```diff
@@ -22,3 +22,12 @@ Windows
...
+| Distribution | Version          | Architecture | WPK Package                                                                                                                                                                                                                                                          |
++==============+==================+==============+======================================================================================================================================================================================================================================================================+
+|    MacOS     | |WAZUH_LATEST|            |    64bit     | `wazuh_agent_v|WAZUH_LATEST|_macos_x86_64.wpk <https://packages.wazuh.com/4.x/wpk/macos/x86_64/pkg/wazuh_agent_v|WAZUH_LATEST|_macos_x86_64.wpk>`_ (`sha512 <https://packages.wazuh.com/|CURRENT_MAJOR|/checksums/wazuh/|WAZUH_LATEST|/wazuh_agent_v|WAZUH_LATEST|_macos_x86_64.wpk.sha512>`__)                      |
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/wpk-list.rst:32]:
```suggestion
|    macOS     | |WAZUH_LATEST|            |    64bit     | `wazuh_agent_v|WAZUH_LATEST|_macos_x86_64.wpk <https://packages.wazuh.com/4.x/wpk/macos/x86_64/pkg/wazuh_agent_v|WAZUH_LATEST|_macos_x86_64.wpk>`_ (`sha512 <https://packages.wazuh.com/|CURRENT_MAJOR|/checksums/wazuh/|WAZUH_LATEST|/wazuh_agent_v|WAZUH_LATEST|_macos_x86_64.wpk.sha512>`__)                      |
```

**Review by MarVergara — CHANGES_REQUESTED** [review]
LGTM after your revision of these suggestions.

File: source/development/packaging/generate-wpk-package.rst [hunk 20]
```diff
@@ -35,12 +35,11 @@ Execute the ``generate_wpk_package.sh`` script, with the different options you d
...
 
   Usage: ./generate_wpk_package.sh [OPTIONS]
-  It is required to use -k or --aws-wpk-key, --aws-wpk-cert parameters
 
-      -t,   --target-system <target>              [Required] Select target wpk to build [linux/windows]
+      -t,   --target-system <target>              [Required] Select target wpk to build [linux/windows/macos].
       -b,   --branch <branch>                     [Required] Select Git branch or tag e.g.
       -d,   --destination <path>                  [Required] Set the destination path of package.
-      -pn,  --package-name <name>                 [Required for windows] Package name to pack on wpk.
+      -pn,  --package-name <name>                 [Required for windows and macos] Package name to pack on wpk.
```
Comment (MarVergara) [inline source/development/packaging/generate-wpk-package.rst:42]:
```suggestion
      -pn,  --package-name <name>                 [Required for windows and macOS] Package name to pack on wpk.
```

File: source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst [hunk 21]
```diff
@@ -133,12 +141,63 @@ Compile the WPK package using the MSI package and, your SSL certificate and key:
...
-    - ``path/to/wazuhagent.msi`` is the path to the MSI you have downloaded in step 3.
-    - ``\*`` is the file (or the files) to be included into the WPK package. In this case, all the contents will be added.
+    - ``path/to/wpkcert.pem`` is the path to the SSL certificate.
+    - ``path/to/wpkcert.key`` is the path to the SSL certificate's key.
+    - ``path/to/wazuhagent.msi`` is the path to the MSI file downloaded in step 3.
+    - ``path/to/upgrade.bat`` is the path to the upgrade.bat file. Find an example in src/win32 in the Wazuh repository.
+    - ``path/to/do_upgrade.ps1`` is the path to the do_upgrade.ps1 file. Find an example in src/win32 in the Wazuh repository.
+
+
+MacOS WPK
```
Comment (MarVergara) [inline source/user-manual/agents/remote-upgrading/create-custom-wpk/generate-wpk-package-manually.rst:151]:
```suggestion
macOS WPK
```

**Review by MarVergara — APPROVED** [review]
LGTM

File: source/development/packaging/generate-wpk-package.rst [hunk 22]
```diff
@@ -63,12 +62,12 @@ Below, you will find an example of Linux WPK package building.
...
   # ./generate_wpk_package.sh -t linux -b v|WAZUH_LATEST| -d /tmp/wpk -k /tmp/keys -o LinuxAgent.wpk
 
-This will build a |WAZUH_LATEST| Wazuh Linux WPK package named LinuxAgent.wpk, using the  with the previously generated keys that are saved in ``/tmp/keys`` and store it in ``/tmp/wpk``.
+This script builds a Wazuh |WAZUH_LATEST| Linux WPK package named LinuxAgent.wpk and stores it in ``/tmp/wpk``. This action is done using the previously generated keys that are saved in ``/tmp/keys``.
 
 Windows WPK
 ^^^^^^^^^^^
 
-To build a WPK for Windows you need to first download an MSI package of the desired version:
+To build a WPK for Windows, you need to first download an MSI package of the desired version:
```
Comment (s-ocando) [inline source/development/packaging/generate-wpk-package.rst:70]:
```suggestion
To build a WPK for Windows, you need to first download a MSI package of the desired version:
```

**Review by s-ocando — APPROVED** [review]
LGTM

**Pull request merged**